### DISCUSSION TITLE
scheduler  exited when the end_date is coming: TypeError: unsupported operand type(s) for +: 'NoneType' and 'datetime.timedelta

### DISCUSSION THREAD

**Discussion by AsusClass** [apache/airflow#19615] [Q&A] [answered]
#### airflow  with 3 schedulers and 6 workers 
![image](https://user-images.githubusercontent.com/28615859/141986922-ad1b734e-d44e-4f00-b2ed-6f91a6be8131.png)

#### DAGs
50 DAGs (10 Tasks for every DAG)  

#### airflow.cfg
![image](https://user-images.githubusercontent.com/28615859/141986987-0cbd65e9-b6d3-43b4-953a-bc702713161e.png)

#### scheduler plan:  run every minute from 19:00~19:30(CST)
![image](https://user-images.githubusercontent.com/28615859/141985702-a963a250-4e80-450b-be0d-27e682bb1a89.png)
**note: the  time in DAGFile Proccessor  is CST but the scheduler is UTC**

#### Error in the last minute from one of schedulers (all the same)
![image](https://user-images.githubusercontent.com/28615859/141985378-737d6106-3b77-423c-b816-fba18777d382.png)

**there should be 500(50*10) succeed tasks in the last minute, but  a few not procced and scheduler dead.**
![image](https://user-images.githubusercontent.com/28615859/141986585-11248f8f-43ed-4552-b2dc-931fa2eb5f59.png)

**Comment by sss12m7** [comment]
啧啧啧\("▔□▔)/\("▔□▔)/\("▔□▔)/

**Reply by AsusClass** [reply]
Maybe soloved in airflow 2.2.2 and i am verifing:
https://github.com/apache/airflow/pull/19307

**Comment by AsusClass** [comment, marked as answer]
This is a bug of airflow and  was soloved in  airflow 2.2.2.

**Comment by 18ksquare24** [comment]
we are getting this error airflow 2.10.1 version, and it is occurring during dags parsing time, we have around 5000 dags
